frontend/src/types/Mesh.ts

```diff
@@ -142,13 +204,41 @@ export interface DecoratedMeshElements {
   nodes?: DecoratedMeshNodeWrapper[];
 }
 
-export type MeshType = 'mesh' | 'node' | 'edge' | 'box';
+export enum MeshType {
+  Box = 'box',
+  Edge = 'edge',
+  Mesh = 'mesh',
+  Node = 'node'
+}
+
+export type BoxNodeData = ClusterNodeData | NamespaceNodeData;
 
-export interface MeshTarget {
-  elem: Controller | GraphElement<ElementModel, any> | undefined;
-  type: MeshType; // the element type
+export interface NodeTarget<T extends MeshNodeData> {
+  elem: Node<NodeModel, T>;
+  type: MeshType.Node;
 }
 
+export interface MeshControllerTarget {
+  elem: Controller | undefined;
+  type: MeshType.Mesh;
+}
+
+export interface EdgeTarget {
+  elem: any;
```

Comment:
If possible, it is better to use unknown, which is the type-safe counterpart of any.

```suggestion
  elem: unknown;
```